src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1385,27 +1491,25 @@ public FormValidation doCheckSshKeysCredentialsId(
          * Tests the connection settings.
          *
          * Overriding needs to {@code @RequirePOST}
-         * @param ec2endpoint
+         * @param region
          * @param useInstanceProfileForCredentials
          * @param credentialsId
          * @param sshKeysCredentialsId
          * @param roleArn
          * @param roleSessionName
-         * @param region
          * @return the validation result
          * @throws IOException
          * @throws ServletException
          */
-        @POST
-        protected FormValidation doTestConnection(
+        @RequirePOST
+        public FormValidation doTestConnection(
```

Comment:
Merging this method with the version in `AmazonEC2Cloud` which just did some prep work and then delegated to this method.